datastream-common/src/main/java/com/linkedin/datastream/metrics/DynamicMetricsManager.java

```diff
@@ -241,11 +246,13 @@ private void updateCache(String simpleClassName, String key, String metricName,
   public void unregisterMetric(String simpleName, String key, String metricName) {
     validateArguments(simpleName, metricName);
     String fullMetricName = MetricRegistry.name(simpleName, key, metricName);
-
+    _registeredMetricRefCount.computeIfPresent(fullMetricName, (localKey, val) -> (val == 1) ? null : val - 1);
```

Comment:
Nitpick: Why do we need computeIfPresent here? Shouldn't we be using compute instead, unless someone is calling unregister without registering a metric?

Reply:
Because MetricRegister.remove handles unregistration without registering a metric.

`/**
     * Removes the metric with the given name.
     *
     * @param name the name of the metric
     * @return whether or not the metric was removed
     */
    public boolean remove(String name) {
        final Metric metric = metrics.remove(name);
        if (metric != null) {
            onMetricRemoved(name, metric);
            return true;
        }
        return false;
    }`